src/App.test.js

```diff
@@ -1,8 +1,70 @@
-import { render, screen } from '@testing-library/react';
+import React from 'react';
+import { shallow } from 'enzyme';
+import * as redux from 'react-redux';
 import App from './App';
+import { contactsMock } from './mocks/contactsMock';
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+  useDispatch: jest.fn(),
+}));
+
+describe('App', () => {
+  const useSelectorMock = redux.useSelector;
+  const useDispatchMock = redux.useDispatch;
+
+  beforeEach(() => {
+    useDispatchMock.mockImplementation(() => () => {});
+  });
+
+  afterEach(() => {
+    useDispatchMock.mockClear();
+    useSelectorMock.mockClear();
+  });
+
+  it('ContactList renders in App if loader is false and error is empty', () => {
+    const mockStore = {
+      contacts: {
+        items: contactsMock,
+        loader: false,
+        error: '',
+      },
+    };
```

Comment:
Please use global store

Reply:
const mockStore has been moved in describe before all tests (it).